src/components/swap/UnsupportedCurrencyFooter.test.tsx

```diff
@@ -0,0 +1,74 @@
+import userEvent from '@testing-library/user-event'
+import { Token } from '@uniswap/sdk-core'
+import { useUnsupportedTokens } from 'hooks/Tokens'
+import { mocked } from 'test-utils/mocked'
+import { fireEvent, render, screen, waitForElementToBeRemoved, within } from 'test-utils/render'
+import { getExplorerLink } from 'utils/getExplorerLink'
+
+import UnsupportedCurrencyFooter from './UnsupportedCurrencyFooter'
+
+const unsupportedTokenAddress = '0x4e83b6287588a96321B2661c5E041845fF7814af'
+const unsupportedTokenSymbol = 'ALTDOM-MAR2021'
+const unsupportedToken = new Token(1, unsupportedTokenAddress, 18, 'ALTDOM-MAR2021')
+const unsupportedTokenExplorerLink = 'www.blahblah.com'
+
+jest.mock('../../hooks/Tokens')
+
+jest.mock('../../utils/getExplorerLink')
+
+describe('UnsupportedCurrencyFooter.tsx with unsupported tokens', () => {
+  beforeEach(() => {
+    mocked(useUnsupportedTokens).mockImplementation(() => ({ [unsupportedTokenAddress]: unsupportedToken }))
+    mocked(getExplorerLink).mockImplementation(() => unsupportedTokenExplorerLink)
+  })
+
+  it('renders', () => {
+    const { asFragment } = render(<UnsupportedCurrencyFooter show={true} currencies={[unsupportedToken]} />)
+    expect(asFragment()).toMatchSnapshot()
+  })
+
+  it('works as expected when one unsupported token exists', async () => {
+    const rendered = render(<UnsupportedCurrencyFooter show={true} currencies={[unsupportedToken]} />)
+    await userEvent.click(screen.getByTestId('read-more-button'))
+    expect(screen.getByText('Unsupported Assets')).toBeInTheDocument()
+    expect(
+      screen.getByText((content) => content.startsWith('Some assets are not available through this interface'))
+    ).toBeInTheDocument()
+    expect(screen.getAllByTestId('unsupported-token-card').length).toBe(1)
+    const unsupportedCard = screen.getByTestId('unsupported-token-card')
+    expect(within(unsupportedCard).getByText(unsupportedTokenSymbol)).toBeInTheDocument()
+    expect(within(unsupportedCard).getByText(unsupportedTokenAddress).closest('a')).toHaveAttribute(
+      'href',
+      unsupportedTokenExplorerLink
+    )
+    await userEvent.click(screen.getByTestId('close-icon'))
+    await waitForElementToBeRemoved(rendered.queryByTestId('unsupported-token-card'))
+    expect(rendered.queryByText('Unsupported Assets')).toBeNull()
+    expect(rendered.queryByTestId('unsupported-token-card')).toBeNull()
+    expect(
+      rendered.queryByText((content) => content.startsWith('Some assets are not available through this interface'))
+    ).toBeNull()
+  }, 10_000)
```

Comment:
Why is this taking so long as to need a timeout? Are you testing something by waiting that could be mocked?

Increasing the timeout is generally a code smell - unit tests should run in under a second, and testing async behavior without mocking them could easily increase the total testing time in the long term.

Reply:
removed fixed, ty